x/authenticator/authenticator/interface.go

```diff
@@ -5,46 +5,68 @@ import (
 )
 
 const (
+	// SignatureVerificationAuthenticator is a Type of authenticator that specific deals with
+	// secp256k1 signatures,
 	SignatureVerificationAuthenticatorType = "SignatureVerificationAuthenticator"
 )
 
+// AuthenticatorData represents the data needed to verify a signer address and message signature
+// TODO: make this less like a void pointer
 type AuthenticatorData interface{}
 
+// Authenticator is an interface used to represent all the authentication functionality needed to
+// verifiy a transaction, pay fees for a transaction and consume gas for verifing a transaction
 type Authenticator interface {
 	// TODO: Rename to DefaultGas or StaticGas. Alt. pass the context and consume the gas here
+	// Gas defines what gas the authenticator uses per signatures verification
 	Gas() uint64
 
+	// Types defines the different types of authenticator, e.g SignatureVerificationAuthenticator
+	// or CosmWasmAuthenticator. Each types of authenticator needs to be registered by the AuthenticatorManager
+	// and these Types are used to store and link the data structure to the Authenticator logic
 	Type() string
 
+	// Initialize is used when an authenticator has been added to the store with
+	// some data, e.g a Public Key but the auth module is not aware of the public key
+	// we then call Initialize on the data, which allows the module to authenticated
+	// a transaction and message
 	Initialize(data []byte) (Authenticator, error)
 
+	// GetAuthenticationData gets signatures and signer data from a transaction
```

Comment:
signature and signer data is specific to SignatureVerificationAuthenticator. In general, this gets any data that needs to be extracted from the tx.

Reply:
Truth, I can update this to be more generic in a later PR if that's ok!

Reply:
Updated to this in the latest:

```
	// GetAuthenticationData gets any authentication data needed from a transaction
	// it returns an interface that is defined as a concrete type by the implementer of the interface.
	// This is used in an ante handler with Authenticate to ensure the user has correct permission to execute
	// a message.
```